mapper-pojo/src/main/java/org/hibernate/search/v6poc/entity/pojo/mapping/definition/annotation/IndexingDependency.java

```diff
@@ -0,0 +1,95 @@
+ * <p>
+ * In particular, some properties might be updated very frequently,
+ * or trigger reindexing to other entities that are very expensive to load in memory.
+ * <p>
+ * See for example this model:
+ *
+ * <pre><code>
+ *     &#064;Indexed
+ *     public class EntityA {
+ *
+ *         &#064;IndexedEmbedded
+ *         &#064;IndexingDependency(reindexOnUpdate = ReindexOnUpdate.NO)
+ *         // + some annotations defining "EntityB.a" as the inverse side of this association
+ *         private EntityB b;
+ *
+ *     }
+ *
+ *     public class EntityB {
+ *         private List&lt;EntityA&gt; a; // This is a very big list
+ *
+ *         &#064;Field
+ *         private String field;
+ *     }
+ * </code></pre>
+ *
+ * Entity A is indexed and embeds entity B.
+ * But the link back from entity B to entity A is a list that might contain a lot of elements,
+ * so we don't want to load this list every time {@code EntityB#field} is updated:
+ * the {@code @IndexingDependency(reindexOnUpdate = ReindexOnUpdate.NO)} annotation prevents that.
+ * This means in particular that updates to {@code EntityB#field}
+ * need to be taken into account through some external process.
+ * One solution would be to reindex every night in a batch process, for example,
+ * which would mean allowing the index to be partly out-of-date for at most 24 hours.
+ */
+@Documented
+@Target({ ElementType.METHOD, ElementType.FIELD })
+@Retention(RetentionPolicy.RUNTIME)
+public @interface IndexingDependency {
+
+	/**
+	 * @return How indexed entities using the annotated property should be reindexed when the value,
+	 * or any nested value, is updated.
+	 * This setting is only effective for values that are actually used when indexing
+	 * (values used in field definitions, in bridges, ...).
+	 * Defaults to {@link ReindexOnUpdate#USED}.
+	 */
+	ReindexOnUpdate reindexOnUpdate() default ReindexOnUpdate.USED;
+
+	/**
+	 * @return An array of reference to container value extractor implementation classes,
+	 * allowing to precisely define which value the indexing process depends on.
+	 * For instance, on a property of type {@code Map<EntityA, EntityB>},
+	 * {@code @AssociationInverseSide(extractors = @ContainerValueExtractorBeanReference(type = MapKeyExtractor.class))}
+	 * would define a dependency to the map keys (of type EntityA),
+	 * while {@code @AssociationInverseSide(extractors = @ContainerValueExtractorBeanReference(type = MapValueExtractor.class))}
+	 * would define a dependency to the map values (of type EntityB).
+	 * By default, Hibernate Search will try to apply a set of extractors for common types
+	 * ({@link Iterable}, {@link java.util.Collection}, {@link java.util.Optional}, ...).
+	 * To prevent Hibernate Search from applying any extractor, set this attribute to an empty array (<code>{}</code>).
+	 */
+	ContainerValueExtractorBeanReference[] extractors()
```

Comment:
I think the javadoc needs some love. The reference to `@AssociationInverseSide` is weird.

Reply:
Done